xorca/lib.py

```diff
@@ -413,3 +414,90 @@ def load_xorca_dataset(data_files=None, aux_files=None, decode_cf=True,
         get_all_compatible_chunk_sizes(target_ds_chunks, ds_xorca))
 
     return ds_xorca
+
+
+def load_xorca_dataset_zarr(data_stores=None, aux_files=None, decode_cf=True,
+                       **kwargs):
+    """Create a grid-aware NEMO dataset from zarr stores.
+
+    Parameters
+    ----------
+    data_stores : Path | sequence | string
+        Zarr stores containing the data: single instance or sequence of
+        anything accepted by `xr.open_zarr`.
+    aux_files : Path | sequence | string
+        Anything accepted by `xr.open_mfdataset` or, `xr.open_dataset`: A
+        single file name, a sequence of Paths or file names, a glob statement.
+    input_ds_chunks : dict
+        Chunks for the ds to be preprocessed.  Pass chunking for any input
+        dimension that might be in the input data.
+    target_ds_chunks : dict
+        Chunks for the final data set.  Pass chunking for any of the likely
+        output dims: `("t", "z_c", "z_l", "y_c", "y_r", "x_c", "x_r")`
+    decode_cf : bool
+        Do we want the CF decoding to be done already?  Default is True.
+
+    Returns
+    -------
+    dataset
+
+    """
+
+    default_input_ds_chunks = {
+        "time_counter": 1, "t": 1,
+        "z": 2, "deptht": 2, "depthu": 2, "depthv": 2, "depthw": 2,
+        "y": 200, "x": 200
+    }
+    # get and remove (pop) the input_ds_chunks from kwargs
+    # to make sure that chunking is not applied again during preprocess_orca
+    input_ds_chunks = kwargs.pop("input_ds_chunks",
+                                 default_input_ds_chunks)
+
+    default_target_ds_chunks = {
+        "t": 1,
+        "z_c": 2, "z_l": 2,
+        "y_c": 200, "y_r": 200,
+        "x_c": 200, "x_r": 200
+    }
+    target_ds_chunks = kwargs.get("target_ds_chunks",
+                                  default_target_ds_chunks)
+
+    # First, read aux files to learn about all dimensions.  Then, open again
+    # and specify chunking for all applicable dims.  It is very important to
+    # already pass the `chunks` arg to `open_[mf]dataset`, to ensure
+    # distributed performance.
+    _aux_files_chunks = map(
+        lambda af: get_all_compatible_chunk_sizes(
+            input_ds_chunks, xr.open_dataset(af, decode_cf=False)),
+        aux_files)
+    aux_ds = xr.Dataset()
+    for af, ac in zip(aux_files, _aux_files_chunks):
+        aux_ds.update(
+            rename_dims(xr.open_dataset(af, decode_cf=False,
+                                        chunks=ac)))
```

Comment:
Ideally, we could also provide Zarr stores for the `aux_files`. A possible way to generalize this would be
```python
    def _open_dataset_or_zarr(*args, **kwargs):
        try:
            return xr.open_dataset(*args, **kwargs)
        except:
            return xr.open_zarr(*args, **kwargs)
        else:
            raise ValueError(
                f"Could not open dataset or zarr with args={args} and kwargs={kwargs}."
            )
            
    _aux_files_chunks = map(
        lambda af: get_all_compatible_chunk_sizes(
            input_ds_chunks, _open_dataset_or_zarr(af, decode_cf=False)),
        aux_files)
    aux_ds = xr.Dataset()
    for af, ac in zip(aux_files, _aux_files_chunks):
        aux_ds.update(
            rename_dims(_open_dataset_or_zarr(
                af, decode_cf=False, chunks=ac
            ))
        )
```

(Not tested.)